chore(android): auto-download NDK r21c for module builds if needed

jquick-axway opened the pull request:
**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-27939
- https://jira.appcelerator.org/browse/TIMOB-27776

**Summary:**
- The macOS distribution of Android NDK r21c has been "notarized" by Google. This means macOS Catalina will no longer show security popups when attempting to do a module build.
- Will only auto-download this NDK version for module builds if no installation was found.
- Fixed builds with NDK r21 to no longer log harmless "Bad file descriptor" errors.

**Module Test:**
1. Use a macOS Catalina machine that does not have the NDK installed.
2. Download the zipped source of module [ti.imagefactory](https://github.com/appcelerator-modules/ti.imagefactory).
3. Open the Terminal.
4. `CD` to directory: `./ti.imagefactory/android`
5. Enter: `appc run -p android`
6. Verify that you see log message similar to the below. This may take a few minutes.
`[INFO]  [GRADLE] Preparing "Install NDK (revision: 21.2.6472646)"`
7. Verify that the module builds and its example project runs.

jquick-axway committed:
chore(android): auto-download NDK r21c for module builds

- This NDK version was notarized for macOS Catalina. Prevents security popups when doing module builds.

build commented:
<!--
  0 failure: 
  0 warning: 
  3 messages
  
  DangerID: danger-id-Danger;
-->



<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-11755/1/artifact/dist/mobilesdk-9.1.0.v20200604172032-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>:fist: The commits in this PR match our conventions! Feel free to Rebase and Merge this PR when ready.</td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 6596 tests are passing.
(There are 702 skipped tests not included in that total)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f54aa897af8b7ed2004e2a4ae4dc4c424004964c
</p>

jquick-axway committed:
fix(android): bad file descriptor errors with NDK v21 builds

- Compiling with Android NDK 21 and higher would cause several harmless "Bad file descriptor" errors to be logged.
- Caused by NDK 21 setting "--output-sync" argument default. Setting it to "none" restores old behavior.

jquick-axway committed:
fix(android): only use --output-sync for NDK 21+

- Using NDK command line argument --output-sync will cause a build failure on NDK 19 and lower. And we don't need to pass it into NDK 20.

jquick-axway committed:
chore(android): removed TIMOB-27939 commented-out code

garymathews committed:
Merge branch 'master' into TIMOB-27939

garymathews reviewed (approved):
CR: **PASS**

ssjsamir reviewed (approved):
**FR Passed:** No longer see `Bad file descriptor` errors and able to install NDK if no other NDKs are found. 

**Test Environment**
```
MacOS Catalina: 10.15.5 Beta
Xcode: 11.4
Java Version: 1.8.0_131
Android NDK: 21.1.6273396-beta2
Node.js: 10.16.3
""NPM":"5.0.0","CLI":"8.0.0""
Andoird Pixl xl 7.1.1 Emulator 
```
